src/vanilla/store.ts

```diff
@@ -514,10 +543,16 @@ const buildStore = (getAtomState: StoreArgs[0]): Store => {
         r = writeAtomState(pending, a, aState, ...args) as R
       }
       flushPending(pending)
+      if (!isSync) {
+        processOnChangeHandlers(pending)
+      }
```

Comment:
This looks bad in the long term maintenance. It should be always paired.
Let's see how we can fix it before working on this PR.